src/components/LayoutTree.tsx

```diff
@@ -378,6 +379,138 @@ const linearizeTree = (
+    ? `left {durations.layoutNodeAnimationFast} ease-out,top {durations.layoutNodeAnimationFast} ease-out`
+    : `left {durations.layoutNodeAnimation} ease-out,top {durations.layoutNodeAnimation} ease-out`
+
+  // when a parent is swapped, we want to animate the curve if the last action is editThought or setCursor
+  const shouldAnimateCurve = isParentSwapped && (isLastActionEditThought || isLastActionSetCursor)
+
+  const outerDivStyle = {
+    // Cannot use transform because it creates a new stacking context, which causes later siblings' DropChild to be covered by previous siblings'.
+    // Unfortunately left causes layout recalculation, so we may want to hoist DropChild into a parent and manually control the position.
+    left: x,
+    top: isLastActionSwapParent ? 0 : y,
+    // Table col1 uses its exact width since cannot extend to the right edge of the screen.
+    // All other thoughts extend to the right edge of the screen. We cannot use width auto as it causes the text to wrap continuously during the counter-indentation animation, which is jarring. Instead, use a fixed width of the available space so that it changes in a stepped fashion as depth changes and the word wrap will not be animated. Use x instead of depth in order to accommodate ancestor tables.
+    // 1em + 10px is an eyeball measurement at font sizes 14 and 18
+    // (Maybe the 10px is from .content padding-left?)
+    width: isTableCol1 && width !== undefined ? width : `calc(100% - ${x}px + 1em + 10px)`,
+    ...(style || {}),
+    textAlign: isTableCol1 ? ('right' as const) : undefined,
+  }
+
+  const innerDivStyle = isLastActionSwapParent
+    ? {
+        top: y,
+        left: 0,
+      }
+    : undefined
+
+  return (
+    <div
+      // The key must be unique to the thought, both in normal view and context view, in case they are both on screen.
+      // It should not be based on editable values such as Path, value, rank, etc, otherwise moving the thought would make it appear to be a completely new thought to React.
+      aria-label='tree-node'
+      className={css({
+        position: 'absolute',
+        transition:
+          isLastActionSwapParent || shouldAnimateCurve
+            ? cursorIsGreaterThanPrevSibling
+              ? isLastActionNewThought
+                ? 'left {durations.layoutNodeAnimationFast} {easings.nodeCurveXLayerClockwise}'
+                : 'left {durations.layoutNodeAnimation} {easings.nodeCurveXLayerClockwise}'
+              : isLastActionNewThought
+                ? 'left {durations.layoutNodeAnimationFast} {easings.nodeCurveXLayer}'
+                : 'left {durations.layoutNodeAnimation} {easings.nodeCurveXLayer}'
+            : transition,
+      })}
+      style={outerDivStyle}
+    >
+      <div
+        className={css({
+          position: 'absolute',
+          transition: isLastActionSwapParent
+            ? cursorIsGreaterThanPrevSibling
+              ? isLastActionNewThought
```

Comment:
The last action cannot be both `swapParent` and `newThought` at the same time. Thus, a nested ternary with these two conditions doesn't make sense.

Reply:
`isLastActionNewThought` is no longer used here.